pkgs/development/haskell-modules/lib/compose.nix

```diff
@@ -418,23 +418,14 @@ rec {
         builtins.listToAttrs (map toKeyVal haskellPaths);
 
   addOptparseApplicativeCompletionScripts = exeName: pkg:
-    builtins.trace "addOptparseApplicativeCompletionScripts is deprecated in favor of generateOptparseApplicativeCompletion. Please change ${pkg.name} to use the latter or its plural form."
-    (generateOptparseApplicativeCompletion exeName pkg);
+    lib.warn "addOptparseApplicativeCompletionScripts is deprecated in favor of haskellPackages.generateOptparseApplicativeCompletions. Please change ${pkg.name} to use the latter and make sure it uses its matching haskell.packages set!"
+    (__generateOptparseApplicativeCompletion exeName pkg);
```

Comment:
I don't feel strongly about this, but maybe we could just remove this function?  It has been here and been deprecated for quite a long time, right?

I can't ever recall seeing this in the wild.

Reply:
Deleting it form `haskell.lib.compose` should be fine in any case, we can still keep it in `haskell.lib` for backwards compatibility.